synapse/util/caches/lrucache.py

```diff
@@ -58,6 +58,18 @@ def __init__(self, max_size, keylen=1, cache_type=dict):
 
         lock = threading.Lock()
 
+        def cache_len():
+            if size_callback is not None:
+                return sum(size_callback(node.value) for node in cache.itervalues())
```

Comment:
This is probably sub-optimal since it iterates the entire cache. You will probably need to store the current size somewhere and update it as things are added or removed.